File: sentimental/caesar.py
import sys

if len(sys.argv) is not 2:  # exit if more than 2 args
    print(f"Usage: {sys.argv[0]} k")
    sys.exit(1)

k = int(sys.argv[1])  # initialize key as int
if k > 26:  # adjust to 0 - 26
    k = (k % 26)

plaintxt = input("plaintext: ")  # plaintext stored as string
strlist = []  # initialize list of char
for i in range(len(plaintxt)):
    strlist.append(plaintxt[i])  # append all char to strlist

for i in range(len(strlist)):
    if strlist[i].isalpha() is True:
        for g in range(k):
            if strlist[i] is 'Z' or strlist[i] is 'z':
                temp = ord(strlist[i]) - 25
                strlist[i] = chr(temp)
            elif strlist[i] >= 'A' or strlist[i] <= 'Z':
                temp = ord(strlist[i]) + 1
                strlist[i] = chr(temp)
            elif strlist[i] >= 'a' or strlist[i] <= 'z':
                temp = ord(strlist[i]) + 1
                strlist[i] = chr(temp)

plaintxt = ''.join(strlist)
print("ciphertext: " + plaintxt)

File: caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

//argc is number of arguments passed
//argv is the input string
int main(int argc, string argv[])
{
    if (argc != 2) //exit if more than 1 command line arg entered
    {
        printf("Usage: %s k\n", argv[0]);
        return 1;
    }

    int k = atoi(argv[1]); //initialize key
    if (k > 26) //adjust to k <= 26
    {
        k = k % 26;
    }

    string plaintxt = get_string("plaintext: "); //take user input
    int length = strlen(plaintxt); //input string length
    for (int i = 0; i < length; i++) //iterate over each char in string
    {
        if (isalpha(plaintxt[i]) != 0)
        {
            for (int g = 0; g < k; g++)
            {
                if ((plaintxt[i] == 'Z') || (plaintxt[i] == 'z'))
                {
                    plaintxt[i] -= 25;
                }
                else if ((plaintxt[i] >= 'A') && (plaintxt[i] < 'Z'))
                {
                    plaintxt[i] += 1;
                }
                else if ((plaintxt[i] >= 'a') && (plaintxt[i] < 'z'))
                {
                    plaintxt[i] += 1;
                }
            }
        }
    }
    printf("ciphertext: %s\n", plaintxt);
    return 0;
}